app/models/server/models/LivechatDepartmentAgents.js

```diff
@@ -11,9 +11,9 @@ export class LivechatDepartmentAgents extends Base {
 		super('livechat_department_agents');
 
 		this.tryEnsureIndex({ departmentId: 1 });
-		this.tryEnsureIndex({ departmentEnabled: 1 });
-		this.tryEnsureIndex({ agentId: 1 });
-		this.tryEnsureIndex({ username: 1 });
+		this.tryEnsureIndex({ departmentEnabled: 1, agentId: 1 });
+		this.tryEnsureIndex({ username: 1, departmentId: 1 });
```

Comment:
instead of creating a new index, I'd change the existing one with `departmentId: 1` to a compound index with `username`.

Reply:
Oh, but I was trying to use the prefixes here:
- username + departmentId will cover queries on both and on username
- departmentEnabled + agentId will cover queries on both
- agentId + departmentId will cover both and agentId only
- departmentId will cover only the ones using departmentId

There's one query that only uses username, so the index may be worth there

We do need some telemetry on index usage on real production envs tho :(

Reply:
maybe my comment was not scoped enough 😬 

I'm talking about only regarding the new index `{ username: 1, departmentId: 1 }`.. because there is no query using only `username` , if you have a compound index with `departmentId` + `username` , it will alone cover both cases:

- a find only using `departmentId`
- a find using both `departmentId` and `username`

since there is already a index for `departmentId` only, I suggested then to change that one to include `username` as well.